osbs/build/user_params.py

```diff
@@ -200,61 +195,42 @@ def set_params(self,
 
         Please keep the paramater list alphabetized for easier tracking of changes
 
-        the following parameters can be pulled from the BuildConfiguration (ie, build_conf)
-        :param build_image: str,
-        :param build_imagestream: str,
+        the following parameters are pulled from the BuildConfiguration (ie, build_conf)
         :param build_from: str,
         :param orchestrator_deadline: int, orchestrator deadline in hours
         :param reactor_config_map: str, name of the config map containing the reactor environment
         :param worker_deadline: int, worker completion deadline in hours
         """
-        if build_conf:
-            build_image = build_conf.get_build_image()
-            build_imagestream = build_conf.get_build_imagestream()
-            build_from = build_conf.get_build_from()
-            self.scratch.value = build_conf.get_scratch(scratch)
-            orchestrator_deadline = build_conf.get_orchestor_deadline()
-            worker_deadline = build_conf.get_worker_deadline()
-            reactor_config_map = reactor_config_map or build_conf.get_reactor_config_map()
-        else:
-            self.scratch.value = scratch
+        if not build_conf:
+            raise OsbsValidationException('build_conf must be defined')
+
+        build_from = build_from or build_conf.get_build_from()
```

Comment:
maybe we can also add build_from to init as BuildParam

and then set that in here, and also use it in plugin_configuration directly without constructing it from build_image / build_imagestream

Reply:
I don't know how much code or complexity it saves, but I went ahead and made the change.

Reply:
looks good